bench/ruby/optimized/iseq.ll

```diff
@@ -2206,19 +2206,20 @@ define dso_local ptr @rb_iseq_new_with_opt(i64 noundef %0, i64 noundef %1, i64 n
 
 36:                                               ; preds = %31, %24
   %37 = phi i16 [ %35, %31 ], [ %30, %24 ]
-  %.mask.i = and i8 %27, 2
-  %38 = icmp eq i8 %.mask.i, 0
-  br i1 %38, label %39, label %set_compile_option_from_ast.exit
-
-39:                                               ; preds = %36
-  %40 = shl i8 %27, 6
-  %41 = zext nneg i8 %40 to i16
-  %42 = and i16 %37, -193
-  %43 = or disjoint i16 %42, %41
-  store i16 %43, ptr %12, align 8
+  %38 = shl i8 %27, 6
+  %39 = ashr exact i8 %38, 6
+  %40 = icmp sgt i8 %39, -1
```

Comment:
Regression. Should be fixed by https://github.com/llvm/llvm-project/pull/141974.